java/ql/src/utils/model-generator/CaptureSummaryModels.ql

```diff
@@ -0,0 +1,280 @@
+  exists(FieldToReturnConfig config, ReturnNodeExt returnNodeExt |
+    config.hasFlow(_, returnNodeExt) and
+    returnNodeExt.getEnclosingCallable() = api and
+    not api.getDeclaringType() instanceof EnumType and
+    isRelevantType(returnNodeExt.getType()) and
+    not (
+      returnNodeExt.getKind() instanceof ValueReturnKind and
+      exists(captureQualifierFlow(api))
+    )
+  |
+    result = asTaintModel(api, "Argument[-1]", asOutput(api, returnNodeExt))
+  )
+}
+
+string asOutput(Callable api, ReturnNodeExt node) {
+  if node.getKind() instanceof ValueReturnKind
+  then result = "ReturnValue"
+  else
+    result = parameterAccess(api.getParameter(node.getKind().(ParamUpdateReturnKind).getPosition()))
+}
+
+class FieldAssignment extends AssignExpr {
+  FieldAssignment() { exists(Field f | f.getAnAccess() = this.getDest()) }
+}
+
+class ParameterToFieldConfig extends TaintTracking::Configuration {
+  ParameterToFieldConfig() { this = "ParameterToFieldConfig" }
+
+  override predicate isSource(DataFlow::Node source) {
+    source instanceof DataFlow::ParameterNode and
+    isRelevantType(source.getType())
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    thisAccess(sink.(DataFlow::PostUpdateNode).getPreUpdateNode())
+  }
+
+  override predicate isAdditionalTaintStep(DataFlow::Node node1, DataFlow::Node node2) {
+    exists(FieldAssignment a |
+      a.getSource() = node1.asExpr() and
+      DataFlow::getFieldQualifier(a.getDest()) = node2.(DataFlow::PostUpdateNode).getPreUpdateNode() and
+      isRelevantType(a.getDest().(FieldAccess).getField().getType())
+    )
+  }
+
+  override DataFlow::FlowFeature getAFeature() {
+    result instanceof DataFlow::FeatureEqualSourceSinkCallContext
+  }
+}
+
+private predicate thisAccess(DataFlow::Node n) {
+  n.asExpr().(InstanceAccess).isOwnInstanceAccess()
+  or
+  n.(DataFlow::ImplicitInstanceAccess).getInstanceAccess() instanceof OwnInstanceAccess
+  or
+  n.asExpr().(FieldAccess).isOwnFieldAccess()
+}
+
+/**
+ * Captures APIs that accept input and store them in a field.
+ * Example:
+ * ```
+ * public class Foo {
+ *  private String tainted;
+ *  public void doSomething(String input) {
+ *    tainted = input;
+ *  }
+ * ```
+ * Captured Model:
+ * `p;Foo;true;doSomething;(String);Argument[0];Argument[-1];taint`
+ */
+string captureFieldFlowIn(Callable api) {
+  exists(DataFlow::Node source, ParameterToFieldConfig config |
+    not api.isStatic() and
+    config.hasFlow(source, _) and
+    source.asParameter().getCallable() = api
+  |
+    result =
+      asTaintModel(api, "Argument[" + source.asParameter().getPosition() + "]", "Argument[-1]")
+  )
+}
+
+class ParameterToReturnValueTaintConfig extends TaintTracking::Configuration {
+  ParameterToReturnValueTaintConfig() { this = "ParameterToReturnValueTaintConfig" }
+
+  override predicate isSource(DataFlow::Node source) {
+    exists(Callable api |
+      source instanceof DataFlow::ParameterNode and
```

Comment:
This line is superfluous - it's implied by `source.asParameter()`. Also, I guess this case isn't intended to include the `this` parameter?